From 81e84200b259bbb06fa88df50e17861da050ded9 Mon Sep 17 00:00:00 2001
From: Elvis Angelaccio <elvis.angelaccio@kde.org>
Date: Wed, 7 Jun 2017 16:55:10 +0200
Subject: [PATCH] Show application icon on Wayland

On Wayland the compositor needs to know the name of the .desktop file in
order to be able to lookup the application icon. Otherwise the
taskmanager and the window title bar show the generic wayland icon as
fallback.

Change-Id: If0a77780a18b7d84aacace74b699b08628b1f097
---
 src/Gui/main.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Gui/main.cpp b/src/Gui/main.cpp
index 76736493a..92a8a0d09 100644
--- a/src/Gui/main.cpp
+++ b/src/Gui/main.cpp
@@ -73,7 +73,9 @@ int main(int argc, char **argv)
     AppVersion::setCoreApplicationData();
 
     app.setAttribute(Qt::AA_UseHighDpiPixmaps);
-
+#if QT_VERSION >= QT_VERSION_CHECK(5, 7, 0)
+    app.setDesktopFileName(QStringLiteral("org.kde.trojita"));
+#endif
     app.setWindowIcon(UiUtils::loadIcon(QStringLiteral("trojita")));
 
     QTextStream qOut(stdout, QIODevice::WriteOnly);
-- 
GitLab

